### PR TITLE
Standardize Factory Footprint Sizes

### PR DESCRIPTION
**Pull request by Tharsy**
This Changes almost every factory in the game; exceptions are factories that don't need changes. 
This makes factories follow a standard size for their tier. The idea of this is to balance the surface area usage between all factions while also allowing the factories to be big enough to support the units they build, however, extended to every factory to fit the following:
T1 = 6x6,
T1 Air = 6x9,
T2 = 9x9,
T3 + T2 Shipyard = 12x12

### Notable changes are:
- T2 factories are now bigger.
- Gantries are bigger and some t1 factories are smaller, but all now align with the grid.
- Armada no longer has smaller labs overall.
- The cortex gantry can fit the Behemoth now!
- The dragon fits in it's factory now.

### Work done
- Rescaled models
- New models for cortex gantry, both t2 air plants, and the Cortex Seaplane Platform.
- Updated scripts for the new Cortex Gantry, both Seaplane Platforms and both of the T2 Air Plants
- Updated unit pictures
- Updated and cleaned up yardmaps for all  related units.
- updated factory ground decals (Actual atlas update in another pr)

### Screenshots:
In the examples below the grid squares on the map are a 3x3 footprint.
image
image

armaap coraap corsy corgant

### Work needed
- Sanity check on anything really

### CODE REVIEW

**Review by WatchTheFort — APPROVED** [review]

File: units/Legion/Labs/legalab.lua
Comment (drivver44) [inline units/Legion/Labs/legalab.lua:1]: hitbox needs adjustment

File: units/Legion/Labs/legvp.lua
Comment (drivver44) [inline units/Legion/Labs/legvp.lua:1]: floating hitbox

File: units/Legion/Labs/legap.lua [hunk 1]
```diff
@@ -16,7 +16,7 @@ return {
 		energystorage = 100,
 		explodeas = "largeBuildingexplosiongeneric",
 		footprintx = 6,
-		footprintz = 5,
+		footprintz = 6,
```
Comment (drivver44) [inline units/Legion/Labs/legap.lua:19]: not 6x9z as listed in the PR standards
`T1 Air = 6x9,`
hitbox needs adjustment as its smaller than the model

File: units/Legion/Labs/legamsub.lua
Comment (drivver44) [inline units/Legion/Labs/legamsub.lua:1]: hitbox is significantly bigger than the model

File: units/Legion/Labs/legavp.lua
Comment (drivver44) [inline units/Legion/Labs/legavp.lua:1]: hitbox needs adjustment
also the model seems off to what the footprint grid should be
the model is slighly off on the sides

File: units/Legion/Labs/leggant.lua
Comment (drivver44) [inline units/Legion/Labs/leggant.lua:1]: hitbox is bigger than model and footprint occupancy

File: units/Legion/Labs/leggant.lua [hunk 2]
```diff
@@ -31,7 +31,7 @@ return {
 		terraformspeed = 3000,
 		usepiececollisionvolumes = 1,
```
Comment (drivver44) [inline units/Legion/Labs/leggant.lua:32]: remove this line
also add hitbox to the unitdef

File: units/Legion/Labs/leglab.lua [hunk 3]
```diff
@@ -31,7 +31,7 @@ return {
 		sightdistance = 290,
 		terraformspeed = 500,
 		workertime = 100,
-		yardmap = "coooocoooooococcococcccooccccocccccc",
+		yardmap = "yooooy oooooo coccoc occcco occcco yccccy",
```
Comment (drivver44) [inline units/Legion/Labs/leglab.lua:34]: remove `y` and replace with `o`
IMO this goes against the standardization your trying to achieve
yes there is a ruling on the footprint occpancy question. i think in this case that ruling should be ignored

File: units/Legion/Labs/leghp.lua
Comment (drivver44) [inline units/Legion/Labs/leghp.lua:1]: hitbox needs adjustment

File: units/Legion/Labs/legjim.lua
Comment (drivver44) [inline units/Legion/Labs/legjim.lua:1]: needs hitbox adjustment

**Review by drivver44 — CHANGES_REQUESTED** [review]
hitbox adjustment can be scaling and offset issue
hitbox scaling just means that the hitbox isnt scaled to the model

floating hitbox means that the hitbox does not intersect the ground

File: units/CorBuildings/LandFactories/coraap.lua [hunk 4]
```diff
@@ -5,15 +5,15 @@ return {
 		buildpic = "CORAAP.DDS",
 		buildtime = 20700,
 		canmove = true,
-		collisionvolumeoffsets = "0 10 -2",
-		collisionvolumescales = "113 43 89",
+		collisionvolumeoffsets = "0 12 0",
+		collisionvolumescales = "128 42 128",
```
[the remark this answers is not in the document]
Comment (drivver44) [reply units/CorBuildings/LandFactories/coraap.lua]: this hitbox is still not occupying the full model and the hitbox does not intersect the ground

File: units/ArmBuildings/LandFactories/armaap.lua [hunk 5]
```diff
@@ -32,7 +32,7 @@ return {
 		sightdistance = 312,
 		terraformspeed = 1000,
 		workertime = 200,
-		yardmap = "oooooooooooooooooooooooooooooooooooooooooooooooo",
+		yardmap = "yyyoooyyy yoooooooy yoooooooy ooooooooo ooooooooo ooooooooo ooooooooo yoooooooy yyoooooyy",
```
Comment (drivver44) [inline units/ArmBuildings/LandFactories/armaap.lua:35]: this should be entirely `o`
if the hitbox occupies the spot so does the footprint
convert the hitbox shape to from `CylY` to box

File: units/Legion/Labs/legaap.lua
Comment (drivver44) [inline units/Legion/Labs/legaap.lua:1]: hitbox does not line up with the coraap hitbox

File: units/ArmBuildings/LandFactories/armalab.lua
Comment (drivver44) [inline units/ArmBuildings/LandFactories/armalab.lua:1]: model is slightly bigger than hibox and footprint occupancy

File: units/ArmBuildings/SeaFactories/armshltxuw.lua
Comment (drivver44) [inline units/ArmBuildings/SeaFactories/armshltxuw.lua:1]: model is bigger than occupied area

File: units/ArmBuildings/SeaFactories/armasy.lua
Comment (drivver44) [inline units/ArmBuildings/SeaFactories/armasy.lua:1]: hitbox does not line up wioth footprint or model occupancy

File: units/CorBuildings/LandFactories/coralab.lua [hunk 6]
```diff
@@ -5,15 +5,15 @@ return {
 		buildpic = "CORALAB.DDS",
 		buildtime = 16800,
 		canmove = true,
-		collisionvolumeoffsets = "0 13 8",
-		collisionvolumescales = "101 51 90",
+		collisionvolumeoffsets = "0 2 0",
+		collisionvolumescales = "140 56 136",
```
Comment (drivver44) [inline units/CorBuildings/LandFactories/coralab.lua:9]: hitbox does not line up with footprint

File: units/CorBuildings/LandFactories/corgant.lua [hunk 7]
```diff
@@ -29,7 +29,7 @@ return {
 		sightdistance = 273,
 		terraformspeed = 3000,
 		workertime = 600,
-		yardmap = "oooooooooooooooooooocccccoooocccccoooocccccoooocccccoooocccccoooocccccoooocccccoo",
+		yardmap = "yyooooooooyy oooooooooooo oooooooooooo ooccccccccoo ooccccccccoo ooccccccccoo ooccccccccoo ooccccccccoo ooccccccccoo ooccccccccoo ooccccccccoo yoccccccccoy",
```
Comment (drivver44) [inline units/CorBuildings/LandFactories/corgant.lua:32]: should be exclusive `o`
if the hitbox occupies a spot so does the footprint

File: units/CorBuildings/LandFactories/coraap.lua [hunk 8]
```diff
@@ -5,15 +5,15 @@ return {
 		buildpic = "CORAAP.DDS",
 		buildtime = 20700,
 		canmove = true,
-		collisionvolumeoffsets = "0 10 -2",
-		collisionvolumescales = "113 43 89",
+		collisionvolumeoffsets = "0 12 0",
+		collisionvolumescales = "140 42 140",
```
Comment (drivver44) [inline units/CorBuildings/LandFactories/coraap.lua:9]: sanity comment 
floating hitbox

File: units/CorBuildings/LandFactories/coravp.lua [hunk 9]
```diff
@@ -31,7 +31,7 @@ return {
 		sightdistance = 286,
 		terraformspeed = 1000,
 		workertime = 300,
-		yardmap = "oooooooooooooooooooocccccoooocccccoooocccccoooocccccoooocccccoo",
+		yardmap = "yoooooooy ooooooooo ooooooooo oocccccoo oocccccoo oocccccoo oocccccoo oocccccoo oocccccoo",
```
Comment (drivver44) [inline units/CorBuildings/LandFactories/coravp.lua:34]: `Y` should be `O`

File: units/ArmBuildings/SeaFactories/armshltxuw.lua
Comment (drivver44) [reply units/ArmBuildings/SeaFactories/armshltxuw.lua:1]: still needs adjusment

File: units/ArmBuildings/LandFactories/armap.lua [hunk 10]
```diff
@@ -32,7 +32,7 @@ return {
 		sightdistance = 292,
 		terraformspeed = 500,
 		workertime = 100,
-		yardmap = "ooooooooo ooooooooo ooooooooo ooooooooo ooooooooo ooooooooo",
+		yardmap = "yoooooooy ooooooooo ooooooooo ooooooooo ooooooooo yoooooooy",
```
Comment (drivver44) [inline units/ArmBuildings/LandFactories/armap.lua:35]: `y` to `o`

File: units/Legion/Labs/legavp.lua [hunk 11]
```diff
@@ -32,7 +32,7 @@ return {
 		sightdistance = 286,
 		terraformspeed = 1000,
 		workertime = 300,
-		yardmap = "oooooooooooooooooooocccccoooocccccoooocccccoooocccccoooocccccoo",
+		yardmap = "yoooooooy ooooooooo ooooooooo oocccccoo oocccccoo oocccccoo oocccccoo oocccccoo oocccccoo",
```
Comment (drivver44) [inline units/Legion/Labs/legavp.lua:35]: `y` to `o`

File: units/Legion/Labs/leggant.lua [hunk 12]
```diff
@@ -29,9 +29,8 @@ return {
 		selfdestructas = "hugeBuildingExplosionGenericSelfd",
 		sightdistance = 273,
 		terraformspeed = 3000,
-		usepiececollisionvolumes = 1,
 		workertime = 600,
-		yardmap = "oooooooooooooooooooocccccoooocccccoooocccccoooocccccoooocccccoooocccccoooocccccoo",
+		yardmap = "yyooooooooyy oooooooooooo oooooooooooo ooccccccccoo ooccccccccoo ooccccccccoo ooccccccccoo ooccccccccoo ooccccccccoo ooccccccccoo ooccccccccoo yoccccccccoy",
```
Comment (drivver44) [inline units/Legion/Labs/leggant.lua:33]: `y` to `o`

**Comment by drivver44** [comment]
safe to merge
any issues that come up are my fault and ill fix @WatchTheFort

**Pull request merged**